helm-v3/005-test.md

```diff
@@ -0,0 +1,27 @@
+# Test Command
+
+## Overview
+
+`helm test` gives a chart author the ability to define actions on a helm release which validate that the chart is running as intended. This is useful for ensuring that a `helm install` or `helm upgrade` created/modified Kubernetes resources successfully and that the consequential `helm release` is functional.
+* A **test** is defined as a command and some arguments executed from a container running in a Kubernetes Pod. If the command completes inside the container with an exit code of 0, it is successful. All other exit codes indicate the test failed.
+* Tests are defined in Manifests that live inside of the `templates/tests/success/` and `templates/tests/failure/` directories inside of a chart.
+  * Tests that live in `templates/tests/success/` are expected to succeed.
+  * Tests that live in `templates/tests/failure` are expected to fail.
+  * In helm 2.x, tests used annotations to denote the expected condition. This will no longer be the case and will be replaced with the aforementioned directory placement.
+* `helm test` commands
+    * `helm test run <release>`
+        * Executes tests by creating the Kubernetes Pods declared in the `templates/tests/success` and `templates/tests/failure` directories of the chart and all subcharts in the `charts/` directory and waits for test results
```

Comment:
Can we address the creation of supporting resources (ie. configmaps, dummy services, service mesh routing rules, etc) that may be needed only during the execution of tests? See https://github.com/helm/helm/issues/5474.

I'd be in favor of annotations over directory structure layouts -- some of these test-only resources may easily need to be shared between multiple test "suites" or jobs/pods.